cluster.go

```diff
@@ -28,94 +25,108 @@ type ClusterClient struct {
 
 // NewClusterClient initializes a new cluster-aware client using given options.
 // A list of seed addresses must be provided.
-func NewClusterClient(opt *ClusterOptions) (*ClusterClient, error) {
-	addrs, err := opt.getAddrSet()
-	if err != nil {
-		return nil, err
-	}
-
+func NewClusterClient(opt *ClusterOptions) *ClusterClient {
 	client := &ClusterClient{
-		addrs:   addrs,
-		conns:   make(map[string]*Client),
+		addrs:   opt.getAddrs(),
+		clients: make(map[string]*Client),
 		opt:     opt,
 		_reload: 1,
 	}
 	client.commandable.process = client.process
-	client.reloadIfDue()
 	go client.reaper(time.NewTicker(5 * time.Minute))
-	return client, nil
+	return client
 }
 
-// Close closes the cluster connection
+// Close closes the cluster client.
 func (c *ClusterClient) Close() error {
-	c.slotsMx.Lock()
-	defer c.slotsMx.Unlock()
-
-	return c.reset()
+	// TODO: close should make client unusable
+	c.setSlots(nil)
+	return nil
 }
 
 // ------------------------------------------------------------------------
 
-// Finds the current master address for a given hash slot
-func (c *ClusterClient) getMasterAddrBySlot(hashSlot int) string {
-	if addrs := c.slots[hashSlot]; len(addrs) > 0 {
-		return addrs[0]
+// getClient returns a Client for a given address.
+func (c *ClusterClient) getClient(addr string) *Client {
+	c.clientsMx.RLock()
+	client, ok := c.clients[addr]
+	if ok {
+		c.clientsMx.RUnlock()
+		return client
 	}
-	return ""
-}
+	c.clientsMx.RUnlock()
 
-// Returns a node's client for a given address
-func (c *ClusterClient) getNodeClientByAddr(addr string) *Client {
-	c.connsMx.Lock()
-	client, ok := c.conns[addr]
+	c.clientsMx.Lock()
+	client, ok = c.clients[addr]
 	if !ok {
 		opt := c.opt.clientOptions()
 		opt.Addr = addr
 		client = NewTCPClient(opt)
-		c.conns[addr] = client
+		c.clients[addr] = client
 	}
-	c.connsMx.Unlock()
+	c.clientsMx.Unlock()
+
 	return client
 }
 
+// randomClient returns a Client for the first live node.
+func (c *ClusterClient) randomClient() (client *Client, err error) {
```

Comment:
What's the advantage of this, compared to storing addrs as a `map[string]struct{}`. Instead of having N attempts, trying to pick a random address (why 10?), it seems cleaner to simply iterate over the addrs map, don't you think?